lib/api/controllers/indexController.js

```diff
@@ -43,166 +36,115 @@ class IndexController extends BaseController {
       'exists',
-      'getAutoRefresh',
       'list',
-      'mDelete',
-      'refresh',
-      'refreshInternal',
-      'setAutoRefresh'
+      'mDelete'
     ]);
-
-    this.engine = this.kuzzle.services.publicStorage;
   }
 
   /**
-   * Reset all indexes
+   * Deletes multiple indexes or all indexes allow to user
    *
-   * @param {Request} request
-   * @returns {Promise<Object>}
-   */
-  mDelete(request) {
-    const allowedIndexes = [];
-
-    assertHasBody(request);
-    assertBodyHasAttribute(request, 'indexes');
-
-    return this.engine.listIndexes()
-      .then(res => {
-        const indexes = res.indexes.filter(
-          index => _.includes(request.input.body.indexes, index)
-        );
-
-        const promises = indexes
-          .map(index => request.context.user
-            .isActionAllowed(
-              new Request(
-                { controller: 'index', action: 'delete', index },
-                request.context),
-              this.kuzzle)
-            .then(isAllowed => {
-              if (isAllowed) {
-                allowedIndexes.push(index);
-              }
-            }));
-
-        return Bluebird.all(promises);
-      })
-      .then(() => {
-        request.input.body.indexes = allowedIndexes;
-        return this.engine.deleteIndexes(request);
-      })
-      .then(response => {
-        response.deleted.forEach(index => this.kuzzle.indexCache.remove(index));
-
-        return response;
-      });
-  }
-
-  /**
-   * Create an empty index
+   * Body:
+   *  - (optional) indexes: String[] - Index names to delete
    *
    * @param {Request} request
-   * @returns {Promise<Object>}
+   *
+   * @returns {Promise.<Object>}
    */
-  create(request) {
-    assertHasIndex(request);
-
-    return this.engine.createIndex(request)
-      .then(response => {
-        this.kuzzle.indexCache.add(request.input.resource.index);
-
-        return response;
-      });
+  mDelete (request) {
+    const indexes = this.getArrayParam(request, 'body.indexes', []);
+
+    return this.publicStorage.listIndexes()
+      .then(publicIndexes => publicIndexes.filter(index => indexes.includes(index)))
+      .then(filtered => this._allowedIndexes(request, filtered))
+      .then(allowed => this.publicStorage.deleteIndexes(allowed))
+      .then(deleted => ({
+        deleted
+      }));
   }
 
   /**
-   * Delete the entire index and associated collections
+   * Creates an index
    *
    * @param {Request} request
-   * @returns {Promise<Object>}
+   *
+   * @returns {Promise.<Object>}
    */
-  delete(request) {
-    assertHasIndex(request);
-
-    return this.engine.deleteIndex(request)
-      .then(response => {
-        this.kuzzle.indexCache.remove(request.input.resource.index);
-
-        return response;
-      });
+  create (request) {
+    const index = this.getIndex(request);
+
+    return this.publicStorage.createIndex(index)
+      .then(() => ({
+        acknowledged: true,
+        shards_acknowledged: true
```

Comment:
I think that this should be removed, either now, or in a future version. If you chose the latter, then a `@todo` mark should be added.